### DISCUSSION TITLE
OPA Policy API

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#385] [OPA and Rego]
Hi There,

I'm currently reading https://www.openpolicyagent.org/docs/latest/rest-api/#policy-api. I also did tests on calling the API, it works.
My questions are why we need policy APIs, what are the unique use cases?
I'm able to create document under any module using data API, so why I need policy api?

Thanks!

**Comment by anderseknert** [comment]
Hi @Ronnie-personal! I think the difference is in the name :) The data API is for CRUD operations on data (including querying), while the policy API is for policies. You can modify documents via the data API, but AFAIK it can't be used to upload policies.
